packages/datagateway-table/src/breadcrumbs.component.tsx

```diff
@@ -0,0 +1,538 @@
+        last: {
+          displayName: 'N/A',
+        },
+      };
+    }
+
+    // Loop through each entry in the path name before the last.
+    // We always skip 2 go ahead in steps of 2 as the we expect
+    // the format to be /{entity}/{entityId}.
+    let hierarchyCount = 0;
+    const pathLength = this.currentPathnames.length;
+    for (let index = 1; index < pathLength; index += 2) {
+      // Get the entity and the data stored on the entity.
+      let entity = this.currentPathnames[index];
+
+      const link = `/${this.currentPathnames.slice(0, index + 3).join('/')}`;
+
+      // Check we are not the end of the pathnames array.
+      if (index < pathLength - 1) {
+        // If the index does not already exist,
+        // create the entry for the index.
+        if (!updatedState.currentHierarchy[hierarchyCount]) {
+          // Insert the new breadcrumb information into the array.
+          updatedState.currentHierarchy.splice(hierarchyCount, 1, {
+            id: '',
+            pathName: entity,
+            displayName: 'N/A',
+            url: link,
+          });
+        }
+
+        // Get the information held in the state for the current path name.
+        const entityInfo = updatedState.currentHierarchy[hierarchyCount];
+
+        // Get the entity Id in the path to compare with the saved one in our state.
+        const entityId = this.currentPathnames[index + 1];
+
+        // Check if an entity id is present or if the id has changed since the last update to the state.
+        if (entityInfo.id.length === 0 || entityInfo.id !== entityId) {
+          // In general the API endpoint will be our entity name and
+          // the entity field we want is the NAME of the entity.
+          let apiEntity = entity;
+          let requestEntityField = 'NAME';
+
+          // There are some exceptions to this when handling the DIAMOND/DLS
+          // depending on if 'proposal' has been found in the current path.
+          if (this.currentPathnames.includes('proposal')) {
+            // If the entity is current proposal, then we will not make an API request,
+            // as we need the investigation ID to retrieve the TITLE.
+            if (entity === 'proposal') {
+              apiEntity = 'investigation';
+              requestEntityField = 'TITLE';
+            } else if (entity === 'investigation') {
+              // Otherwise, we can proceed and get the VISIT_ID for the investigation entity.
+              requestEntityField = 'VISIT_ID';
+            }
+          } else {
+            // Anything else (including ISIS), we request the TITLE for the investigation entity.
+            if (entity === 'investigation') requestEntityField = 'TITLE';
+          }
+
+          // Create the entity url to request the name, this is pluralised to get the API endpoint.
+          let requestEntityUrl;
+          if (entity !== 'proposal') {
+            requestEntityUrl = `${apiEntity}s`.toLowerCase() + `/${entityId}`;
```

Comment:
internationalisation

Reply:
actually, is this? this is a url - are these internationalised in the current topcat? to keep the urls the same?

Reply:
@louise-davies does this need to be changed? This is an API call, does it need to be internationalised?

Reply:
Ah, if it's an API call then no. I just searched for `}s` and commented on all of them. This one is fine.